src/checks/series/step_check.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{DataCache, Error, Flag};
+
+/// Number of leading values a [`DataCache`] must contain to QC all its
+/// intended values with step check
+pub const STEP_LEADING_PER_RUN: u8 = 1;
+
+/// Timeseries QC test that compares each observation against its immediate predecessor.
+///
+/// If the absolute value of the difference between the observed value and it's predecessor is
+/// greater than max, Flag::Fail will be returned for that observation, if greater than high,
+/// Flag::Warn, if either value if missing, Flag::DataMissing, else Flag::Pass.
+///
```

Comment:
Maybe we can be a bit more systematic with the docs? Something like
```suggestion
/// Timeseries QC test that compares each observation against its immediate predecessor.
/// 
/// Returns:
/// - [`Flag::DataMissing`], if either value is missing,
/// - [`Flag::Warn`], if the difference is greater than `high`,
/// - [`Flag::Fail`], if the difference is greater than `max`,
/// - [`Flag::Pass`], otherwise.
///
```
 and keep the same style for every check (as much as possible, of course).
Also, did we decide not to return `Flag::Warn`?

Reply:
done